tox.ini

```diff
@@ -0,0 +1,86 @@
+# This file is managed by bootstack-charms-spec and should not be modified
+# within individual charm repos. https://launchpad.net/bootstack-charms-spec
+
+[tox]
+skipsdist=True
+envlist = lint, unit, func
+
+[testenv]
+basepython = python3
+setenv =
+  PYTHONPATH = {toxinidir}:{toxinidir}/lib/:{toxinidir}/hooks/
+passenv =
+  HOME
+  PATH
+  CHARM_*
+  PYTEST_KEEP_MODEL
+  PYTEST_CLOUD_NAME
+  PYTEST_CLOUD_REGION
+  PYTEST_MODEL
+  MODEL_SETTINGS
+  HTTP_PROXY
+  HTTPS_PROXY
+  NO_PROXY
+  SNAP_HTTP_PROXY
+  SNAP_HTTPS_PROXY
+  OS_*
+
+[testenv:dev-environment]
+envdir = {toxinidir}/.venv
+deps =
+  pre-commit
+  {[testenv:lint]deps}
+  {[testenv:unit]deps}
+  ; {[testenv:func]deps} # temporarily comment it out until func tests are added
```

Comment:
I think we can add structure for func tests (folder, empty requirements, bundles folder, etc.) or use empty `deps` and warning message in `commands` . Because like this `dev-environment` is working but `tox -e func` will failed and `make funct` also.

Reply:
Done